### PR TITLE
Improve Facebook creatives design

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Pat wants them to look as much as possible to Guardian Frame ads, so I added the "Ad" pill on the top right and changed the typography accordingly.

picture 1

cc @JonNorman @jbreckmckye @rich-nguyen @Calanthe

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js [hunk 1]
```diff
@@ -1,63 +1,84 @@
...
-            this.$adSlot[0].insertAdjacentHTML('beforeend', markup);
+                FB.Event.subscribe(
+                    'ad.error',
+                    function(errorCode, errorMessage, placementID) {
+                        reportError(new Error('Facebook returned an empty ad response'), {
+                            feature: 'commercial',
+                            placementID: placementID,
+                            errorMessage: errorMessage
+                        }, false);
+                        resolve();
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js]: Shouldn't this `reject`, so we can short circuit the `addLabel()` and `callback[...]` flows?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js]: Yep, good point.

File: static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js [hunk 2]
```diff
@@ -1,63 +1,84 @@
...
+                    }
+                );
+            });
 
-            if (this.params.trackingPixel) {
-                addTrackingPixel(this.$adSlot, this.params.trackingPixel + this.params.cacheBuster);
-            }
+            var markup = facebookTpl(Object.assign({ externalLink: svgs('externalLink') }, adUnits[this.params.placement]));
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js]: `Object.assign` is ES6, so it won't work on IE11 - amongst others.

File: static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js [hunk 3]
```diff
@@ -454,6 +454,8 @@ define([
             if (mobileAdSizes && mobileAdSizes.indexOf(adSizes.stickyMpu) > -1) {
                 stickyMpu($adSlot);
             }
+        } else if ($adSlot.hasClass('ad-slot--facebook')) {
+            $adSlot.addClass('ad-slot--fluid');
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js]: Given the way that `then` operations are run a tick after the promise they listen to resolves, will there be a frame where the advert appears without its requisite styling? Perhaps we could just use `ad-slot--fluid` right from the start?

File: static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js [hunk 4]
```diff
@@ -621,6 +623,7 @@ define([
     function shouldRenderLabel($adSlot) {
         return !$adSlot.hasClass('ad-slot--frame') &&
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js]: Is there a better way to communicate whether a component handles its own labelling? We are having to swap between different classes on the same advert just to control this behaviour.

File: static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js (hunk 3 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js]: Good point, I had to verify but no. The promise resolves when the callback is called, which can only happen after the elements has been added to the DOM.

File: static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js (hunk 4 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js]: Yes there is, but this is a bigger change that will have ramifications in other creatives. If you want take a stab at it 😃

File: static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js]: damn, good spot. I meant to change that.

File: static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js (hunk 3 above)
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js]: That's the problem, though - if the elements are added to the DOM with `ad-slot--facebook`, they won't be full width on their initial render. I'm not sure if users will be able to perceive this in practice, though.

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js]: The whole `callback` thing needs to be rethought, that's for sure.

**Comment by jbreckmckye** [comment]
Great stuff 👍

**Pull request merged**